src/service/standard-actions-impl.js

```diff
@@ -163,6 +174,74 @@ export class StandardActions {
     throw user().createError('Unknown AMP action ', method);
   }
 
+  /**
+   * Handles the copy to clipboard action
+   * @param {!./action-impl.ActionInvocation} invocation
+   * @return {!null}
+   */
+  handleCopy_(invocation) {
+    /** @enum {string} */
+    const CopyEvents = {
+      COPY_ERROR: 'copy-error',
+      COPY_SUCCESS: 'copy-success',
+      SUPPORT_ERROR: 'support-error',
+    };
+    let eventName;
+    const {args, node} = invocation;
+    const win = getWin(node);
+
+    let textToCopy;
+    if (invocation.tagOrTarget === 'AMP') {
+      /**
+       * Copy Static Text
+       *  Example: AMP.copy(text='TextToCopy');
+       */
+      textToCopy = args['text'].trim();
+    } else {
+      /**
+       * Copy Target Element Text
+       *  Example: targetId.copy();
+       */
+      const target = devAssertElement(invocation.node);
+      textToCopy = (target.value ?? target.textContent).trim();
+    }
+
+    /**
+     * Trigger Event based on copy action
+     *  - If content got copied to the clipboard successfully, it will
+     *  fire `copy-success` event.
+     *  - If there's any error in copying, it will
+     *  fire `copy-error` event.
+     *  - If browser is not supporting the copy function/action, it
+     *  will fire `support-error` event.
+     *
+     *  Example: <button on="tap:AMP.copy(text='Hello AMP');copy-success:copied.show()">Copy</button>
+     */
+    if (isCopyingToClipboardSupported(win.document)) {
+      if (copyTextToClipboard(win, textToCopy)) {
+        eventName = CopyEvents.COPY_SUCCESS;
+      } else {
+        eventName = CopyEvents.COPY_ERROR;
+      }
+    } else {
+      eventName = CopyEvents.SUPPORT_ERROR;
```

Comment:
Can these two be collapsed? If the browser doesn't support the copy action, we can fire `copy-error` with `event.data.type = 'browser'` to indicate a support error.

Reply:
This has been fixed.